File: frontend/lib/core/utils/booking_utils.dart
List<Map<String, String>> getDailyEntries(Map<String, dynamic> booking) {
  final start = DateTime.tryParse(booking['event_start_date'] ?? '');
  final end = DateTime.tryParse(booking['event_end_date'] ?? '');
  final startTime = booking['start_time'] ?? '';
  final endTime = booking['end_time'] ?? '';

  if (start == null || end == null) return [];

  final entries = <Map<String, String>>[];
  DateTime current = start;

  while (!current.isAfter(end)) {
    entries.add({
      'date': current.toIso8601String().split('T').first,
      'start_time': startTime,
      'end_time': endTime,
    });
    current = current.add(const Duration(days: 1));
  }

  return entries;
}

File: backend/utils/bookingUtils.js
function getDailyEntries(booking) {
  const entries = [];
  const start = new Date(booking.event_start_date);
  const end = new Date(booking.event_end_date);
  const startTime = booking.start_time;
  const endTime = booking.end_time;

  for (
    let date = new Date(start);
    date <= end;
    date.setDate(date.getDate() + 1)
  ) {
    entries.push({
      date: date.toISOString().split('T')[0],
      start_time: startTime,
      end_time: endTime,
    });
  }

  return entries;
}
